### ISSUE TITLE
iOS eval() not returning results in UIWebView or WKWebView

### ISSUE THREAD

**Issue by nolanlawson** [appium/appium#6363]
## The problem

Running `/execute` (`browser.eval()`) in the WebView context returns no results. Here's a snippet of the logs:

```
[debug] [RemoteDebugger] Received applicationSentData response
[debug] [RemoteDebugger] Got a blank data response from debugger
```

^ This is sent multiple times, to the point where my console fills up with logs.

The same code works on Android (6.0) , but fails on iOS. I'm using the simulator with iOS 9.1 on Yosemite.

## Environment

* Appium version (or git revision) that exhibits the issue: 1.5.0
* Last Appium version that did not exhibit the issue (if applicable): N/A
* Desktop OS/version used to run Appium: i386 MacBookAir6,2 Darwin, Yosemite 10.10.5
* Node.js version (unless using Appium.app|exe): 5.9.0
* Mobile platform/version under test: iOS 9.1 Simulator for iPhone 6
* Real device or emulator/simulator: Simulator
* Appium CLI or Appium.app|exe: wd 0.4.0

## Details

I'm writing a JavaScript library for hybrid apps (Android/iOS) with JavaScript-based tests. The only web driver API that I need to work is "evaluate JavaScript."

## Link to Appium logs

~~See snippet above. I can't provide the full logs because they're enormous. Also see full steps to reproduce below.~~

Got the full logs for ya: https://gist.github.com/nolanlawson/9e5bdd9248a1d5410d8aa91ded2164bb
Note the part where it says `Received data response from socket send: '{"status":0,"value":...`. That's the object I want returned. Then note the part that says `Responding to client with driver.execute() result: []`. That's the empty object I actually get. :cry: 

## Code To Reproduce Issue

My project is open-source, so I've set up a permanent branch you can test on.

```bash
git clone --branch perma-repro https://github.com/nolanlawson/sqlite-plugin-2.git
cd sqlite-plugin-2
npm install
```

Then to test against UIWebView (reproduces the bug):

```bash
npm run test-ios
```

Or to test against WKWebView (also reproduces the bug):

```bash
WKWEBVIEW=1 npm run test-ios
```

Or to test against Android (does _not_ reproduce the bug - Android is working fine):

```bash
npm run test-android
```

**Comment by scottdixon** [comment]
Thanks for the detailed report.

I believe @imurchie published a fixed for this in 1.5.1 (see [changelog](https://github.com/appium/appium/blob/master/CHANGELOG.md)). Please update + let me know if the issue persists.

**Comment by imurchie** [comment]
I'm checking to see if it works for me with the latest stuff.

**Comment by imurchie** [comment]
Yes. There was an error in handling return values from `executeScript`, which caused zuul to have problems. This has been fixed in Appium 1.5.1.

**Comment by nolanlawson** [comment]
Sweet, this is fixed in 1.5.1! Thank you!!!! :tada:
